### PR TITLE
feat: [DEL-3609]: move delegateListing with token api to ng manager

### PR DESCRIPTION
**Pull request by arpit3107**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31795)

### CODE REVIEW

File: 120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateTokenNgResource.java [hunk 1]
```diff
@@ -152,4 +151,30 @@ public DelegateTokenNgResource(DelegateTokenNgClient delegateTokenClient, Access
     return new RestResponse<>(RestClientUtils.getResponse(
         delegateTokenClient.getTokens(accountIdentifier, orgIdentifier, projectIdentifier, status)));
   }
+
+  @GET
+  @Path("/delegate-token")
```
Comment (gaurav-jain1) [inline 120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateTokenNgResource.java:156]: Api : **/delegate-token-ng/delegate-token** seems  confusing, consider renaming it to '/delegate-token-ng/delegate-groups'

File: 400-rest/src/main/java/io/harness/delegate/resources/DelegateNgTokenInternalResource.java [hunk 2]
```diff
@@ -103,4 +110,21 @@ public DelegateNgTokenInternalResource(DelegateNgTokenService delegateTokenServi
     DelegateEntityOwner owner = DelegateEntityOwnerHelper.buildOwner(orgIdentifier, projectIdentifier);
     return new RestResponse<>(delegateTokenService.getDelegateTokens(accountIdentifier, owner, status));
   }
+
+  @GET
+  @Path("/delegate-token")
```
Comment (gaurav-jain1) [inline 400-rest/src/main/java/io/harness/delegate/resources/DelegateNgTokenInternalResource.java:115]: delegate-groups

**Review by gaurav-jain1 — COMMENTED** [review]
Looks good, minor comments inline.

**Review by anilchowdhury-harness — DISMISSED** [review]
:lgtm:

> Reviewed 1 of 4 files at r1, 2 of 2 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31795#-:-MzExCsCAoRnMZCd--Vm:bnfp4nl)* status: :shipit: complete! ✓

**Review by gaurav-jain1 — COMMENTED** [review]
> Reviewed 1 of 4 files at r1, 2 of 2 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31795)* status: :shipit: complete! ✓

**Review by anilchowdhury-harness — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31795#-:-MzJdEWp3tVg7IlxxrGQ:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**